src/Core/Math/Math.hpp

```diff
@@ -73,10 +73,10 @@ template <typename T>
 inline constexpr T lerp( const T& a, const T& b, Scalar t );
 
 /// As define by https://registry.khronos.org/OpenGL-Refpages/gl4/html/smoothstep.xhtml
-template <typename T, typename U, typename V>
-V smoothstep( T edge0, U edge1, V x ) {
+template <typename T>
```

Comment:
Why removing 2 template parameters ? Should have one for edges, one for the data. This last might be the same than the first, or not.

Reply:
the current implementation I propose could not handle smoostep( scalar, scalar, vector ) nor smoostep(vector, vector, vector) due to operation and cast undefined.
One way to handle this is to have a template specialization, if you have a possible implementation in mind, feel free to test. 

So to ensure the smoothstep works as expected, let's say edge and x are the same time for the moment.

Reply:
I'll give a try ASAP (say, 1 or 2 days).

Reply:
well, compile and works using llvm clang++ 11.1.0
```
template <typename T, template<typename, int... > typename M, int... p>
M<T, p...> smoothstep( T edge0, T edge1, M<T, p...> v ) {
    return v.unaryExpr( [edge0, edge1]( T x ) { return Ra::Core::Math::smoothstep( edge0, edge1, x ); } );
}
```

tested with : 
```
Vector3 v {0_ra, 0_ra, 0_ra};
std::cout << smoothstep(-1_ra, 1_ra, v ).transpose() << "\n"; 
```
this results in `0.5 0.5 0.5` (expected)

and 
```
Matrix2 m;
m << 0_ra, 0_ra, 0_ra, 0_ra;
std::cout << smoothstep(-1_ra, 1_ra, m ) << "\n";
```
this results in (expected also)
```
0.5 0.5
0.5 0.5
```

Reply:
I can push my proposal.

Reply:
pushed